### PR TITLE
ci: add kubernetes helm chart and gcp deployment

### PR DESCRIPTION
**Pull request by gabreal**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -77,25 +79,30 @@ check:runtime:
...
     TARGET: native
+  # <<:                              *kubernetes_build
+  tags:
+    - linux-docker
   only:
     - tags
     - master
     - schedules
     - web
-    - /^pr-[0-9]+$/
```
Comment (gnunicorn) [inline .gitlab-ci.yml]: not on PRs anymore?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -242,10 +256,59 @@ publish:s3:doc:
...
 
 
+
+
+.deploy:template:                  &deploy
+  stage:                           deploy
+  when:                            manual
+  cache:                           {}
+  retry:                           1
+  image:                           dtzar/helm-kubectl:$HELM_VERSION
```
Comment (gnunicorn) [inline .gitlab-ci.yml]: I understand this is a microsoft developer, however, is it okay with our security policies to use a personal third-party docker-image to deploy production?

//cc @kirushik

File: .gitlab-ci.yml (hunk 1 above)
Comment (gabreal) [reply .gitlab-ci.yml]: synching between github and gitlab has been changed and pull requests cannot be prepended with `pr-` string anymore. That's why there is the new one below.

File: .gitlab-ci.yml (hunk 2 above)
Comment (kirushik) [reply .gitlab-ci.yml]: @gnunicorn well, actually it's a valid concern.
I don't think we have a general policy here — but maybe forking https://github.com/dtzar/helm-kubectl to paritytech, setting up DockerHub autobuilds and using the resulting image would be enough? The `Dockerfile` in that repo seems to be a low-maintenance thing.

Comment (gabreal) [reply .gitlab-ci.yml]: we now have our own: https://hub.docker.com/r/parity/kubectl-helm

**Pull request merged**